# ![Genalice](images/GENALICE_COR_logo-RGB.jpg)

# Introduction
This repository describes Materials & Methods for CNV calling with Genalice software against a _benchmark_ dataset: the ICR96 Exon Validation series.

### ICR96 Exon Validation series
This study makes use of the ICR96 Exon CNV Validation series data generated by Professor Nazneen Rahmanâ€™s team at The Institute of Cancer Research, London as part of the TGMI. This dataset consists of 96 samples for orthogonal assessment of exon CNV calling in Next Generation Sequencing (NGS) data. It includes high-quality sequencing data from a targeted NGS assay ([the TruSight Cancer Panel](https://www.illumina.com/Documents/products/datasheets/datasheet_trusight_cancer.pdf)) together with Multiplex Ligation-dependent Probe Amplification (MLPA) results for 96 independent samples.

The dataset has been developed by The Institute of Cancer Research and is described in [this publication](https://wellcomeopenresearch.org/articles/2-35/v1).

### Genalice
Genalice solutions process NGS datasets through read alignment and variant calling, while delivering significant speed increases and accuracy improvements. Genalice delivers solutions for various NGS secondary analyses using innovative and patented techonology, which is further described on our website: [genalice.com](http://www.genalice.com).

# Preprint
This work has been published as a preprint.

__Validation of an ultra-fast CNV calling tool for Next Generation Sequencing data using MLPA-verified copy number alterations.__ Bas Tolhuis and Hans Karten, bioRXive. doi:...

# Quick links
* [Materials](Materials.md)

* [Methods](Methods.md)

# Contact
You can raise issues on this Github repository or contact us: info@genalice.com.

# History
    * June 2018 - Initial release
